app/src/main/java/com/nervousfish/nervousfish/modules/database/GsonDatabaseAdapterHelper.java

```diff
@@ -0,0 +1,86 @@
+package com.nervousfish.nervousfish.modules.database;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.reflect.TypeToken;
+import com.nervousfish.nervousfish.data_objects.DatabasePass;
+import com.nervousfish.nervousfish.data_objects.IKey;
+import com.nervousfish.nervousfish.modules.cryptography.EncryptedSaver;
+import com.nervousfish.nervousfish.modules.filesystem.IFileSystem;
+import com.nervousfish.nervousfish.service_locator.IServiceLocator;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.Serializable;
+import java.lang.reflect.Type;
+
+/**
+ * Helper method for the logical functionality of the {@link GsonDatabaseAdapter}.
+ */
+class GsonDatabaseAdapterHelper implements Serializable {
```

Comment:
I think that the name is too vague. Can you rename this class to GsonDatabaseAdapterLoader, move the contents of the method loadDatabase from GsonDatabaseAdapter to this class and let the old method call this new one?